src/openms/source/FORMAT/MSPGenericFile.cpp

```diff
@@ -160,53 +166,57 @@ namespace OpenMS
         throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
                                             "The current spectrum misses the Name information.");
       }
-      output_file << "Name: " << spectrum.getName() << '\n';
-      const auto& synonyms = spectrum.getMetaValue("Synon");
-      if (synonyms.valueType() == OpenMS::DataValue::DataType::STRING_VALUE)
+      if (spectrum.size())// we will not store spectrum with no peaks
       {
-        size_t start;
-        size_t end = 0;
-        std::string synonyms_str(synonyms);
-        std::string separator(synonyms_separator_);
-        while ((start = synonyms_str.find_first_not_of(separator, end)) != std::string::npos)
+        output_file << "Name: " << spectrum.getName() << '\n';
+        output_file << "Retention Time: " << spectrum.getRT() << '\n';
+        const auto& synonyms = spectrum.getMetaValue("Synon");
+        if (synonyms.valueType() == OpenMS::DataValue::DataType::STRING_VALUE)
         {
-          end = synonyms_str.find(separator, start);
-          output_file << "Synon: " << synonyms_str.substr(start, end - start) << '\n';
+          size_t start;
+          size_t end = 0;
+          std::string synonyms_str(synonyms);
+          std::string separator(synonyms_separator_);
+          while ((start = synonyms_str.find_first_not_of(separator, end)) != std::string::npos)
+          {
+            end = synonyms_str.find(separator, start);
+            output_file << "Synon: " << synonyms_str.substr(start, end - start) << '\n';
+          }
```

Comment:
I can see this is efficient, but a bit boilerplate.
If the `synonyms_separator_` where a char, one could do
```
  StringList list;
  synonyms.split(synonyms_separator_, list);
  for (const auto& syn : list) output_file << "Synon: " << syn << '\n';
```

but its a string unfortunately. But from taking a quick look at the param_:
`"The character that will separate the synonyms in the Synon metaValue."`
the synonyms_separator_ member could simply be a char.